src/JSInterop/Microsoft.JSInterop/src/DotNetDispatcher.cs

```diff
@@ -276,16 +334,48 @@ private static (MethodInfo, Type[]) GetCachedMethodInfo(string assemblyName, str
             return result;
         }
 
-        private static Assembly GetRequiredLoadedAssembly(string assemblyName)
+        private static Assembly GetRequiredLoadedAssembly(AssemblyKey assemblyKey)
         {
             // We don't want to load assemblies on demand here, because we don't necessarily trust
             // "assemblyName" to be something the developer intended to load. So only pick from the
             // set of already-loaded assemblies.
             // In some edge cases this might force developers to explicitly call something on the
             // target assembly (from .NET) before they can invoke its allowed methods from JS.
             var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies();
-            return loadedAssemblies.FirstOrDefault(a => a.GetName().Name.Equals(assemblyName, StringComparison.Ordinal))
-                ?? throw new ArgumentException($"There is no loaded assembly with the name '{assemblyName}'.");
+            return loadedAssemblies.FirstOrDefault(a => new AssemblyKey(a).Equals(assemblyKey))
+                ?? throw new ArgumentException($"There is no loaded assembly with the name '{assemblyKey.AssemblyName}'.");
         }
+
+        private readonly struct AssemblyKey : IEquatable<AssemblyKey>
+        {
+            public AssemblyKey(Assembly assembly)
+            {
+                Assembly = assembly;
+                AssemblyName = assembly.GetName().Name;
+            }
+
+            public AssemblyKey(string assemblyName)
+            {
+                Assembly = null;
+                AssemblyName = assemblyName;
+            }
+
+            public Assembly Assembly { get; }
+
+            public string AssemblyName { get; }
+
+            public bool Equals(AssemblyKey other)
+            {
+                if (Assembly != null && other.Assembly != null)
```

Comment:
Is this condition backwards by mistake? It looks like you're trying to avoid property dereferences if either one is null, but instead it *only* dereferences the properties if either one is null.

Reply:
I mean, if `other == null`, then we'll skip the `if` block. Then on L374, evaluating `other.AssemblyName` will throw a nullref.

Reply:
These are value types, so the instances (`other`) wouldn't be null. It's just trying to see if both items have the `Assembly` property set at which point we compare those.